### ISSUE TITLE
Execute-ProcessAsUser failing with -wait language translation problem

### ISSUE THREAD

**Issue by FRacine** [PSAppDeployToolkit/PSAppDeployToolkit#228]
Hi,

I get the following error with Execute-ProcessAsUser.
```
Select-Object : La propriété « Status » est introuvable.
Au caractère E:\Temp\Test1.0_Frv1\_Include\AppDeployToolkit\AppDeployToolkitMain.ps1:4652 : 105
+ ... V) | ConvertFrom-CSV | Select-Object -ExpandProperty 'Status' | Selec ...
+                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument : (@{Nom de l'h“te...s en cours=N/A}:PSObject) [Select-Object], PSArgumentException
    + FullyQualifiedErrorId : ExpandPropertyNotFound,Microsoft.PowerShell.Commands.SelectObjectCommand
 
```

The problem is linked to that command line:

`While ((($exeSchTasksResult = & $exeSchTasks /query /TN $schTaskName /V /FO CSV) | ConvertFrom-CSV | Select-Object -ExpandProperty 'Status' | Select-Object -First 1) -eq 'Running') {`

I am only seeing that problem with the -wait parameter. 
I saw that problem on Windows 8.1 and 10 and did not tried it on other OS.

I am running PSADTK 3.6.8.

The problem seems to be we are running french OS. 
So in English, it is "Status" but in french it is "Statut".

2016-12-06 'Running' should be read as 'En cours' in french.

Also in the ame function:

`[int32]$executeProcessAsUserExitCode = ($exeSchTasksResult = & $exeSchTasks /query /TN $schTaskName /V /FO CSV) | ConvertFrom-CSV | Select-Object -ExpandProperty 'Last Result' | Select-Object -First 1
`
I get this error:
```
Select-Object : La propriété « Dernier résultat » est introuvable.
Au caractère E:\Temp\Test1.0_Frv1\_Include\AppDeployToolkit\AppDeployToolkitMain.ps1:4657 : 145
+ ... rtFrom-CSV | Select-Object -ExpandProperty 'Dernier résultat' | Selec ...
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument : (@{Nom de l'h“te...s en cours=N/A}:PSObject) [Select-Object], PSArgumentException
    + FullyQualifiedErrorId : ExpandPropertyNotFound,Microsoft.PowerShell.Commands.SelectObjectCommand


```
Should be 

`[int32]$executeProcessAsUserExitCode = ($exeSchTasksResult = & $exeSchTasks /query /TN $schTaskName /V /FO CSV) | ConvertFrom-CSV | Select-Object -ExpandProperty "Dernier r‚sultat" | Select-Object -First 1
`
Last result = Dernier résultat in french but in running we see  "Dernier résultat" was replace by "Dernier r‚sultat"

Notice the double-quote instead the single quote as delimiter for "Dernier r‚sultat"
The é was replace by , by Windows as it does not like accent character.

Not sure how it might be solve to be generic.

Thanks,

François

**Comment by amotaz** [comment]
I don't know that there's  much we can do about this, however, I am curious about if someone else has run into this running this on other non-english language systems. It seems like the kind of thing that would have cropped up before. Does anyone who uses the toolkit in a non-english setting have any feedback they can provide?

**Comment by FRacine** [comment]
Hi,

About your question, it dépends of how many peoples are using it on non-English oS and if they are using this wonderful function. :) The function will be always working but never wait. I suppose most people will be using it while creating activesetup.

Would it be possible to store those translate values Inside the XML and using them instead hardcoding the text?

I ran:
C:\WINDOWS\System32\schtasks.exe /query /TN $schTaskName /V /FO list

and got

```
Dossierÿ: \
Nom de l'h“te:                                             xxxxx
Nom de la tƒche:                                           \PSAppDeployToolkit-ExecuteAsUser
Prochaine ex‚cution:                                       N/A
**Statut:**                                                    Prˆt
Mode d'ouverture de session:                               Interactive uniquement
Heure de la derniŠre ex‚cution:                            2016-12-28 07:22:56
**Dernier r‚sultat**:                                          0
Auteur:                                                    N/A
Tƒche … ex‚cuter:                                          mytask
D‚marrer dans:                                             N/A
Commentaire:                                               N/A
Statut de la tƒche planifi‚e:                              Activ‚e
Dur‚e d'inactivit‚:                                        D‚sactiv‚e
Gestion de l'alimentation:                                 
Ex‚cuter en tant qu'utilisateur:                           xxxxxxxx
Supprimer la tƒche si elle n'est pas replanifi‚e:          D‚sactiv‚e
Arrˆter la tƒche aprŠs X heures et X minutes:              72:00:00
Planification:                                             Les donn‚es de planification ne sont pas disponibles dans ce format.
Type de planification:                                     Sur demande seulement
Heure de d‚but:                                            N/A
Date de d‚but:                                             N/A
Date de fin:                                               N/A
Jours:                                                     N/A
Mois:                                                      N/A
R‚p‚terÿ: Chaque:                                          N/A
R‚p‚terÿ: Jusqu'…ÿ: Heure:                                 N/A
R‚p‚titionÿ: pendant une dur‚e donn‚e:                     N/A
R‚p‚titionÿ: arrˆter si l'ex‚cution est toujours en cours: N/A
```

**Comment by DennisVH** [comment]
ComObject 'Schedule.Service' doesn't seem to be OS language dependent.

Suggested solution is add a function to retrieve the scheduled task properties (f.e. in the begin block of the Execute-ProcessAsUser function):

```
Function Get-ScheduledTask_EPAU
{
  Param (
    [string]$TaskName
  )

  $Schedule = New-Object -ComObject 'Schedule.Service'
  $Schedule.Connect()
  $RootFolder = $Schedule.GetFolder('\')
  $Task = $RootFolder.GetTask("$TaskName")
  Write-Output $Task
}
```


replace
`While ((($exeSchTasksResult = & $exeSchTasks /query /TN $schTaskName /V /FO CSV) | ConvertFrom-CSV | Select-Object -ExpandProperty 'Status' | Select-Object -First 1) -eq 'Running') {`
with
`while ((Get-ScheduledTask_EPAU -TaskName "\$schTaskName").State -eq 4) {`

and replace
`[int32]$executeProcessAsUserExitCode = ($exeSchTasksResult = & $exeSchTasks /query /TN $schTaskName /V /FO CSV) | ConvertFrom-CSV | Select-Object -ExpandProperty 'Last Result' | Select-Object -First 1`
with
`[int32]$executeProcessAsUserExitCode = (Get-ScheduledTask_EPAU -TaskName "\$schTaskName").LastTaskResult`

I have tested on a Windows 7 x64 where Display Language is Japanese for both User and System accounts.

**Comment by FRacine** [comment]
It is working on Windows 8.1 X64 French canadian.

**Comment by FRacine** [comment]
Any news with this issue?

**Comment by FRacine** [comment]
Will a solution be implement in the next revision of psadtk?

**Comment by FRacine** [comment]
Any followup on this issue?

**Comment by seanlillis** [comment]
See #321 - we will look to include this proposed fix in the next release.

**Comment by DennisVH** [comment]
This will be fixed in the next release for Windows Vista and higher

**Comment by cribban** [comment]
This is the same issue as I reported way back in issue #96 and I saw DennisVH have put that as resolved for next release. Solution is to use a COM object and look at an integer instead of string to make it language neutral. (To solve it for now follow my fix in issue #96)

**Comment by seanlillis** [comment]
This is fixed in v3.7.0.
